### PR TITLE
Add .Net6 RC1 bits

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: build/constants.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     dot-net-core-31-sdk-version: 3.1.412
     dot-net-core-50-sdk-version: 5.0.302
     dot-net-50-sdk-version: 5.0.400
-    dot-net-60-sdk-version: 6.0.100-preview.7.21379.14
+    dot-net-60-sdk-version: 6.0.100-rc.1
```
Comment (qianz2) [inline build/constants.yaml:29]: the full version is 6.0.100-rc.1.21458.32

File: build/__dotNetCoreSdkVersions.sh [hunk 2]
```diff
@@ -8,3 +8,4 @@ DOT_NET_CORE_31_SDK_VERSION='3.1.412'
 DOT_NET_CORE_50_SDK_VERSION='5.0.302'
 DOT_NET_50_SDK_VERSION='5.0.400'
 DOT_NET_60_SDK_VERSION='6.0.100-preview.7.21379.14'
```
Comment (qianz2) [inline build/__dotNetCoreSdkVersions.sh:10]: DOT_NET_60_SDK_VERSION='6.0.100-rc.1.21458.32'

File: platforms/dotnet/versionsToBuild.txt [hunk 3]
```diff
@@ -58,4 +58,6 @@
 
 6.0.100-preview.7.21379.14, c8757325407b5eb1e3870f0db87eeaf44df978313c0b2a7f465ec7d0a5647317cba597264ec81577ea0b3bd97bd33d782234392e8e592e073126792a0406df7b,
 
+6.0.100-rc.1, 7a20c5ac3f8ae7a5a67b0ddc184e7eb845b57aacea65e42e451c51dd48afb8c854f25a879777f0067617bba131ad0c09fb395c62b3ec292915f9434213523463,
```
Comment (qianz2) [inline platforms/dotnet/versionsToBuild.txt:61]: same here

File: tests/Oryx.BuildImage.Tests/CommandTests/PlatformsCommandTest.cs [hunk 4]
```diff
@@ -58,6 +58,7 @@ public void ListsPlatformsAndVersionsAvailableForDynamicInstall()
                     Assert.True(dotNetCorePlatform.Versions.Contains("1.1.13"));
                     Assert.True(dotNetCorePlatform.Versions.Contains("5.0.0-preview.3.20214.6"));
                     Assert.True(dotNetCorePlatform.Versions.Contains("6.0.0-preview.1.21102.12"));
+                    Assert.True(dotNetCorePlatform.Versions.Contains("6.0.100-rc.1"));
```
Comment (qianz2) [inline tests/Oryx.BuildImage.Tests/CommandTests/PlatformsCommandTest.cs:61]: same here, we use full version here

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreSdkVersions.cs [hunk 5]
```diff
@@ -12,5 +12,6 @@ public static class DotNetCoreSdkVersions
         public const string DotNetCore50SdkVersion = "5.0.302";
         public const string DotNet50SdkVersion = "5.0.400";
         public const string DotNet60SdkVersion = "6.0.100-preview.7.21379.14";
+        public const string DotNet6Rc1SdkVersion = "6.0.100-rc.1";
```
Comment (qianz2) [inline src/BuildScriptGenerator/DotNetCore/DotNetCoreSdkVersions.cs]: public const string DotNet60SdkVersion = "6.0.100-rc.1.21458.32";
And remove DotNet6Rc1SdkVersion

File: build/__dotNetCoreSdkVersions.sh (hunk 2 above)
Comment (william-msft) [reply build/__dotNetCoreSdkVersions.sh:10]: Thank you, updated

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs [hunk 6]
```diff
@@ -26,9 +26,9 @@ public static class DotNetCoreRunTimeVersions
         public const string NetCoreApp50Sha = "454a11d87199be6fcd5f97f28b0f9c2f15c0288dd0759dc96ac2c730697b9e6514a19bbfc06185ccb4289dd11681ea5e9cfaf6f3676040e36d21e9b90282af8d";
         public const string AspnetCoreApp50 = "5.0.9";
         public const string AspnetCoreApp50Sha = "67ac99b8f631684961aa2e887ed5d6b9cc301b06a1842ac212791860bf6de145831228df5f57eca8a97e16e569e7b5a308fd20d82f11569c375c689fc932dfff";
-        public const string NetCoreApp60 = "6.0.0-preview.7.21377.19";
-        public const string NetCoreApp60Sha = "959ef85c0cb5c9aa8dab90f5e09d9f9216d546a32b5338f389787523b9c4b48b911ac42a4e6f8c9e1b656bc8f0173ff93beed12439e3f87a52b84d810969969a";
-        public const string AspnetCoreApp60 = "6.0.0-preview.7.21378.6";
-        public const string AspnetCoreApp60Sha = "24e8626d8a2bd126a786ecf97d68e705571567357f6b7370a2b80ac42eff2b1b6e14734be4c0317e2902ea01f2903b285b1e669e4ae76f80ae876dfbf9f0d0c3";
+        public const string NetCoreApp60 = "6.0.100-rc.1.21458.32";
+        public const string NetCoreApp60Sha = "7a20c5ac3f8ae7a5a67b0ddc184e7eb845b57aacea65e42e451c51dd48afb8c854f25a879777f0067617bba131ad0c09fb395c62b3ec292915f9434213523463";
```
Comment (qianz2) [inline src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs:30]: I think you update with SDK ones, we should use the runtime one instead:
public const string NetCoreApp60 = "6.0.0-rc.1.21451.13";
        public const string NetCoreApp60Sha = "6c101a93766747deddc97dd3ae9cbd0d0babc292dff2897006e1df8ad6af100aa74063d4961d73e67c82db66662f760cf83f510617559a2538210fc89694d9ea";

File: build/__dotNetCoreRunTimeVersions.sh [hunk 7]
```diff
@@ -22,7 +22,7 @@ NET_CORE_APP_50='5.0.9'
 NET_CORE_APP_50_SHA='454a11d87199be6fcd5f97f28b0f9c2f15c0288dd0759dc96ac2c730697b9e6514a19bbfc06185ccb4289dd11681ea5e9cfaf6f3676040e36d21e9b90282af8d'
 ASPNET_CORE_APP_50='5.0.9'
 ASPNET_CORE_APP_50_SHA='67ac99b8f631684961aa2e887ed5d6b9cc301b06a1842ac212791860bf6de145831228df5f57eca8a97e16e569e7b5a308fd20d82f11569c375c689fc932dfff'
-NET_CORE_APP_60='6.0.0-preview.7.21377.19'
-NET_CORE_APP_60_SHA='959ef85c0cb5c9aa8dab90f5e09d9f9216d546a32b5338f389787523b9c4b48b911ac42a4e6f8c9e1b656bc8f0173ff93beed12439e3f87a52b84d810969969a'
-ASPNET_CORE_APP_60='6.0.0-preview.7.21378.6'
-ASPNET_CORE_APP_60_SHA='24e8626d8a2bd126a786ecf97d68e705571567357f6b7370a2b80ac42eff2b1b6e14734be4c0317e2902ea01f2903b285b1e669e4ae76f80ae876dfbf9f0d0c3'
+NET_CORE_APP_60='6.0.100-rc.1.21458.32'
+NET_CORE_APP_60_SHA='7a20c5ac3f8ae7a5a67b0ddc184e7eb845b57aacea65e42e451c51dd48afb8c854f25a879777f0067617bba131ad0c09fb395c62b3ec292915f9434213523463'
```
Comment (qianz2) [inline build/__dotNetCoreRunTimeVersions.sh:26]: Same here, this is SDK version not runtime version and sha.

File: tests/Oryx.BuildImage.Tests/CommandTests/PlatformsCommandTest.cs (hunk 4 above)
Comment (william-msft) [reply tests/Oryx.BuildImage.Tests/CommandTests/PlatformsCommandTest.cs]: Thank you updated

File: platforms/dotnet/versionsToBuild.txt (hunk 3 above)
Comment (william-msft) [reply platforms/dotnet/versionsToBuild.txt]: Updated, thank you

File: build/constants.yaml (hunk 1 above)
Comment (william-msft) [reply build/constants.yaml]: Updated, thank you

File: build/constants.yaml [hunk 8]
```diff
@@ -65,10 +65,11 @@
     aspnet-core-app-50: 5.0.9
     aspnet-core-app-50-sha: 67ac99b8f631684961aa2e887ed5d6b9cc301b06a1842ac212791860bf6de145831228df5f57eca8a97e16e569e7b5a308fd20d82f11569c375c689fc932dfff
 
-    net-core-app-60: 6.0.0-preview.7.21377.19
-    net-core-app-60-sha: 959ef85c0cb5c9aa8dab90f5e09d9f9216d546a32b5338f389787523b9c4b48b911ac42a4e6f8c9e1b656bc8f0173ff93beed12439e3f87a52b84d810969969a
-    aspnet-core-app-60: 6.0.0-preview.7.21378.6
-    aspnet-core-app-60-sha: 24e8626d8a2bd126a786ecf97d68e705571567357f6b7370a2b80ac42eff2b1b6e14734be4c0317e2902ea01f2903b285b1e669e4ae76f80ae876dfbf9f0d0c3
+    net-core-app-60: 6.0.100-rc.1.21458.32
+    net-core-app-60-sha: 7a20c5ac3f8ae7a5a67b0ddc184e7eb845b57aacea65e42e451c51dd48afb8c854f25a879777f0067617bba131ad0c09fb395c62b3ec292915f9434213523463
```
Comment (qianz2) [inline build/constants.yaml:69]: same here.

File: build/__dotNetCoreRunTimeVersions.sh (hunk 7 above)
Comment (william-msft) [reply build/__dotNetCoreRunTimeVersions.sh:26]: Thank you, updated

File: build/constants.yaml (hunk 8 above)
Comment (william-msft) [reply build/constants.yaml:69]: Thank you, updated

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs (hunk 6 above)
Comment (william-msft) [reply src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs:30]: Thank you for catching this, updated

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreSdkVersions.cs (hunk 5 above)
Comment (william-msft) [reply src/BuildScriptGenerator/DotNetCore/DotNetCoreSdkVersions.cs]: Updated, thanks

**Review by qianz2 — APPROVED** [review]

File: build/constants.yaml [hunk 9]
```diff
@@ -36,7 +36,7 @@
       file-name-prefix: __
 - name: dot-net-core-run-time-versions
   constants:
-    dot-net-core-runtime-base-tag: 20210825.1
+    dot-net-core-runtime-base-tag: 20210917.2
```
Comment (qianz2) [inline build/constants.yaml:39]: 20210920.2

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs [hunk 10]
```diff
@@ -4,7 +4,7 @@ namespace Microsoft.Oryx.BuildScriptGenerator.DotNetCore
 {
     public static class DotNetCoreRunTimeVersions
     {
-        public const string DotNetCoreRuntimeBaseTag = "20210825.1";
+        public const string DotNetCoreRuntimeBaseTag = "20210917.2";
```
Comment (qianz2) [inline src/BuildScriptGenerator/DotNetCore/DotNetCoreRunTimeVersions.cs:7]: 20210920.2

**Review by qianz2 — APPROVED** [review]

**Review by gpcastro — APPROVED** [review]

**Pull request merged**